app/renderer/helpers/connectionStore.js

```diff
@@ -33,17 +33,14 @@ export const saveConnection = ({
   }
 }
 
-export const updateConnection = (uid, { name, address, username, password }) => {
-  const newConnectionId = getHash(name + address)
-  if (!isExist(newConnectionId)) {
```

Comment:
This is actually necessary. I guess you removed this because you weren't  able to update username and password? The reason is that each connection has unique id which created from the name + address and since they didn't change and the if rule was resulted with negative condition. In order to solve it you need to add the user name and the password as params to the getHash function both on creating method and the update one. This is probably not an ideal solution as more and more params will be added in the future, but for now lets go with that solution, unless you got better idea.